mlrun/api/api/endpoints/marketplace.py

```diff
@@ -78,7 +81,29 @@ async def list_sources(
         auth_info,
     )
 
-    return await run_in_threadpool(get_db().list_marketplace_sources, db_session)
+    marketplace_sources = await run_in_threadpool(
+        get_db().list_marketplace_sources, db_session
+    )
+    # filter sources by item-name, tag or version if given
+    if item_name:
+        filtered_sources = []
+        for source in marketplace_sources:
+            catalog = await run_in_threadpool(
+                mlrun.api.crud.Marketplace().get_source_catalog,
+                source.source,
+                version,
+                tag,
+            )
+            for item in catalog.catalog:
+                if item.metadata.name == item_name:
+                    filtered_sources.append(source)
+                    break
+        return filtered_sources
+    elif tag or version:
+        raise mlrun.errors.MLRunBadRequestError(
+            "Tag or version params must be provided with item-name param"
+        )
+    return marketplace_sources
```

Comment:
Don't do this filtering here in the endpoint - please move it to the crud's `list_marketplace_sources` function (pass the parameters there and perform the filtering). The api endpoint should not do domain-specific functionality.